cmd/processesChecker.go

```diff
@@ -0,0 +1,56 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+)
+
+// WarnMultipleProcesses warns if there are multiple AzCopy processes running
+func WarnMultipleProcesses(directory string, currentPid int) error {
+	currPidFileName := strconv.Itoa(currentPid) + ".pid"
+
+	pidsSubDir := path.Join(directory, "pids") // Made subdir to not clog main dir
+	if err := os.MkdirAll(pidsSubDir, 0755); err != nil {
+		glcm.Error(fmt.Sprintf("error creating pids dir: %v", err))
+	}
+	filePath := path.Join(pidsSubDir, currPidFileName) // E.g "\.azcopy\pids\\XXX.pid"
+	// Creates .pid file with specific pid
+	file, err := os.OpenFile(filePath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
+	if err != nil {
+		glcm.Error(fmt.Sprintf("error creating the .pid file: %v", err))
+		return err
+	}
+	defer file.Close()
+
+	// Check if there is a matching .pid file in directory
+	pidFileRegex, _ := regexp.Compile(`\.pid$`)
+	dir, err := os.ReadDir(pidsSubDir)
+	if err != nil {
+		glcm.Error(fmt.Sprintf("error reading dir: %v", err))
+		return err
+	}
+	for _, fileName := range dir {
```

Comment:
If we're gonna have the regex anyway, I think it's worth calling [os.FindProcess](https://pkg.go.dev/os#FindProcess) to check that the PID is still valid and alive (and cleaning the pid file up if not), otherwise, if AzCopy panics and doesn't exit gracefully, we're still guaranteed to clean it up on future runs.

It's not a _perfect_ fix to the problem, because what if another process takes up residence on the same PID?

At the very least, this'd prevent a customer from being left scratching their head after AzCopy crashed and they just. Eternally got this warning for the rest of time until they figured out the pids folder was the problem.